components/AuthView/AuthView.js

```diff
@@ -0,0 +1,373 @@
+                  this.setState({
+                    register: true,
+                    password: '',
+                    username: '',
+                    passwordConfirm: '',
+                  });
+                }}>
+                <Text style={styles.toggleText}>Create A New Account?</Text>
+              </TouchableOpacity>
+            </View>
+          )}
+          {!this.state.register ? null : (
+            <View style={styles.container}>
+              <Image
+                source={{
+                  uri:
+                    'https://i.ya-webdesign.com/images/research-vector-clinical-trial-2.png',
+                }}
+                style={styles.logo}
+              />
+
+              <TextInput
+                value={this.state.username}
+                onChangeText={(username) => this.setState({username})}
+                placeholder="Username"
+                placeholderTextColor="#cccccc"
+                style={styles.input}
+              />
+
+              <TextInput
+                value={this.state.email}
+                onChangeText={(email) => this.setState({email})}
+                placeholder="Email"
+                placeholderTextColor="#cccccc"
+                style={styles.input}
+              />
+
+              <TextInput
+                value={this.state.password}
+                onChangeText={(password) => this.setState({password})}
+                placeholder="Password"
+                placeholderTextColor="#cccccc"
+                secureTextEntry
+                style={styles.input}
+              />
+
+              <TextInput
+                value={this.state.passwordConfirm}
+                onChangeText={(passwordConfirm) =>
+                  this.setState({passwordConfirm})
+                }
+                placeholder="Confirm Password"
+                placeholderTextColor="#cccccc"
+                secureTextEntry
+                style={styles.input}
+              />
+
+              {this.state.regError ? (
+                <Text style={styles.errText}>{this.state.regError}</Text>
+              ) : null}
+
+              <TouchableOpacity
+                style={styles.loginButton}
+                onPress={(e) => {
+                  this.register(e);
+                }}>
+                <Text style={styles.loginButtonText}>Register</Text>
+              </TouchableOpacity>
+              <TouchableOpacity
+                style={styles.toggle}
+                onPress={(e) => {
+                  this.setState({
+                    register: false,
+                    password: '',
+                    username: '',
+                    passwordConfirm: '',
+                  });
+                }}>
+                <Text style={styles.toggleText}>Already Have An Account?</Text>
+              </TouchableOpacity>
+            </View>
+          )}
+        </View>
+      </TouchableWithoutFeedback>
+    );
+  }
+}
+
+const styles = StyleSheet.create({
```

Comment:
One quick way to shorten this file would be to `export default StyleSheet.create(...)` from a `styles.js` file right beside the `AuthView.js` one.